src/operator/mshadow_op.h

```diff
@@ -80,6 +80,30 @@ struct tanh_grad {
   }
 };
 
+struct exp {
+  MSHADOW_XINLINE static real_t Map(real_t a) {
+    return expf(a);
+  }
+};
+
+struct exp_grad {
+  MSHADOW_XINLINE static real_t Map(real_t a) {
```

Comment:
can I move square_root_grad and square_grad from src/ndarray/unary_function-inl.h to src/operator/mshadow_op.h  ?